code/controllers/subsystem/icon_smooth.dm

```diff
@@ -17,10 +17,24 @@ SUBSYSTEM_DEF(icon_smooth)
 		smooth_queue_cache.len--
 		if(QDELETED(smoothing_atom) || !(smoothing_atom.smoothing_flags & SMOOTH_QUEUED))
 			continue
-		if(smoothing_atom.flags_1 & INITIALIZED_1)
+
+		var/defer = FALSE
+		if(!(smoothing_atom.flags_1 & INITIALIZED_1))
+			defer = TRUE
+		else if(isturf(smoothing_atom))
+			var/turf/smoothing_turf = smoothing_atom
+			// So, when we go to load a template, we wipe the turfs information to prepare it for loading from the template
+			// Which is fine, and you would think nothing could go wrong here
+			// However, if one of the turfs bordering the border of the template area gets triggered to smooth,
+			// it tries to access the loading turfs canSmoothWith list, which will be null!
+			if(smoothing_turf.turf_flags & TURF_MAPLOADING)
+				defer = TRUE
+
+		if(!defer)
 			smoothing_atom.smooth_icon()
 		else
 			deferred += smoothing_atom
```

Comment:
would it be worth just duplicating the deferred removal and using continues to avoid needing to create a new var here?

Reply:
It's not that hot of a proc, however if you think it's worth it I can probably figure something out